tket2-py/tket2/ops.py

```diff
@@ -49,13 +76,21 @@ def __eq__(self, other: object) -> bool:
 
 
 class Pauli(Enum):
-    """Simple enum representation of Pauli matrices."""
+    """Simple enum representation of Pauli matrices.
+
+    Implements the `ToCustomOp` protocol.
+    """
 
     I = auto()  # noqa: E741
     X = auto()
     Y = auto()
     Z = auto()
 
+    def to_custom(self) -> CustomOp:
+        extension_name = "quantum.tket2"
+        gate_name = self.name
+        return CustomOp(extension_name, gate_name, [QB_T], [QB_T])
```

Comment:
The tests that cover Tk2Op (`test_pauli_prop.py`) currently uses the `Pauli*` commands definition, which is a different thing from these.

I specifically did not touch the commands in this PR since they are getting reworked once we start using the hugr builder.

This method should get some coverage once we get there :)